index.dd

```diff
@@ -98,8 +96,31 @@ void main()
 ----
 )
 
+) $(COMMENT your-code-here)
+))) $(COMMENT intro, div, div)
+
+$(COMMENT
+Picking the frontpage example after the page has loaded leads to reflows
+and re-rendering. Hence, we pick the example as soon as possible
+)
+$(SCRIPT
+    // Searches any object for the first hit.
+    function findFirst(el, fn) {
+        return Array.prototype.filter.call(el, fn)[0];
+    }
+    (function() {
+    // randomly pick one example and bootstrap the runnable editor
+    var examples = document.getElementsByClassName('your-code-here-extra');
+    var rouletteIndex = Math.floor(Math.random() * examples.length);
+    var rouletteChild = examples[rouletteIndex];
+
+    // step 1: select the .runnable-example <div>
+    var rouletteChildNode = findFirst(rouletteChild.children, function(e) { return e.className.indexOf("runnable-example") != -1; });
+    // step 2: select the first <pre> element (of the .runnable-example <div>)
+    var el = findFirst(rouletteChildNode.children, function(e) { return e.nodeName == "PRE";});
```

Comment:
Also, I believe you can avoid traversing the element's children by setting a class on the `rouletteChild` element.